addons/website_sale_loyalty/tests/test_shop_multi_reward.py

```diff
@@ -48,22 +52,64 @@ def test_claim_reward_with_multi_product(self):
                 'reward_product_tag_id': tag.id,
                 'reward_product_qty': 1,
                 'required_points': 1,
-            })]
+            })],
         })
 
-        website = self.env['website'].browse(1)
-        order = self.env['sale.order'].create({
-            'website_id': website.id,
-            'partner_id': partner.id,
+        cls.env['loyalty.generate.wizard'].with_context(active_id=cls.coupon_program.id).new({
+            'coupon_qty': 1,
+            'points_granted': 2,
+        }).generate_coupons()
+        cls.coupon = cls.coupon_program.coupon_ids
+
+        cls.cart = cls.env['sale.order'].create({
+            'website_id': cls.website.id,
+            'partner_id': cls.partner.id,
             'order_line': [Command.create({
-                'product_id': product1.id,
+                'product_id': cls.product1.id,
                 'product_uom_qty': 1,
             })],
+            'applied_coupon_ids': [Command.link(cls.coupon.id)],
         })
-        order._update_programs_and_rewards()
-        with MockRequest(self.env, website=website, sale_order_id=order.id):
+        cls.cart._update_programs_and_rewards()
+
+        installed_modules = set(cls.env['ir.module.module'].search([
+            ('state', '=', 'installed'),
+        ]).mapped('name'))
+        for _ in http._generate_routing_rules(installed_modules, nodb_only=False):
+            pass
 
-            WebsiteSaleController.claim_reward(promo_program.reward_ids[:1].id, product_id=str(product2.id))
+    def test_claim_reward_with_multi_product(self):
+        order = self.cart
+        product2 = self.product2
+
+        with MockRequest(self.env, website=self.website, sale_order_id=order.id):
+            self.WebsiteSaleController.claim_reward(
+                self.coupon_program.reward_ids[:1].id,
+                product_id=str(product2.id),
+                code=self.coupon.code,
+            )
 
             self.assertEqual(len(order.order_line), 2, 'reward line should be added to order')
             self.assertEqual(order.order_line[1].product_id, product2, 'added reward line should should contain product 2')
+
+    def test_claim_multi_reward_with_multi_product(self):
```

Comment:
```suggestion
    def test_claim_reward_with_multi_products_among_multi_rewards(self):
```
or something :shrug: :sweat_smile:

Reply:
Thinking about it, this one doesn't even test the `multi_product` reward, so just `test_apply_coupon_with_multiple_rewards` should be adequate.